bin/logging.ts

```diff
@@ -18,9 +23,60 @@ export function setLogThreshold(threshold: LogLevel) {
   logThreshold = threshold;
 }
 
-export function log(level: LogLevel, message: string) {
+export function setGlobalProgressListener(listener: IPublishProgressListener) {
+  globalProgressListener = listener;
+}
+
+// Convert log level to event type
+function logLevelToEventType(level: LogLevel): EventType {
+  switch (level) {
+    case 'error':
+      return EventType.FAIL;
+    case 'verbose':
+      return EventType.DEBUG;
+    default:
+      return EventType.DEBUG;
+  }
+}
+
+export function log(level: LogLevel, message: string, percentComplete?: number) {
   if (LOG_LEVELS[level] >= LOG_LEVELS[logThreshold]) {
-    // eslint-disable-next-line no-console
     console.error(`${level.padEnd(7, ' ')}: ${message}`);
+
+    // Write to progress listener if configured
+    if (globalProgressListener) {
+      const progressEvent: IPublishProgress = {
+        message: `${message}`,
+        percentComplete: percentComplete,
+        abort: () => {},
+      };
+      globalProgressListener.onPublishEvent(logLevelToEventType(level), progressEvent);
```

Comment:
So this is sending the log message to the `globalProgressListener`?
Which in turn when using `DefaultProgressListener ` then calls `log`?
How does that work?

Reply:
This is needless right now, but I decided to build it this way as I wanted to provide an easy angle by which someone could add progress listeners even if using the cli. It's a bit pre-emptive so I can remove it if we want as it's un-needed complexity at this point.

Reply:
If you are removing progress listeners, what is left in this PR?

My point here was the loop you are creating:

`log()` -> `globalProgressListener.onPublishEvent()` -> `DefaultProgressListener` -> `log()`

Reply:
It's also not tested. Pretty sure if you had a test with `DefaultProgressListener` you would notice that.

Reply:
Ended up doing basically a rewrite so this isn't relevant anymore